src/test/java/com/app/onlinebookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import com.app.onlinebookstore.model.Book;
+import com.app.onlinebookstore.repository.BookRepository;
+import com.app.onlinebookstore.service.impl.BookServiceImpl;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    private static Book book;
+    private static BookDto bookDto;
+    private static BookDtoWithoutCategoryIds bookDtoWithoutCategoryIds;
+    private static final Long TEST_ID = 1L;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @BeforeAll
+    public static void setUp() {
+        book = new Book();
+        book.setId(TEST_ID);
+        book.setTitle("Test Title");
+        book.setAuthor("Test Author");
+        book.setIsbn("1234567890123");
+        book.setPrice(new BigDecimal("19.99"));
+        book.setDescription("Test Description");
+        book.setCoverImage("Test Cover Image");
+
+        bookDto = new BookDto();
+        bookDto.setId(book.getId());
+        bookDto.setTitle(book.getTitle());
+        bookDto.setAuthor(book.getAuthor());
+        bookDto.setIsbn(book.getIsbn());
+        bookDto.setPrice(book.getPrice());
+        bookDto.setDescription(book.getDescription());
+        bookDto.setCoverImage(book.getCoverImage());
+
+        bookDtoWithoutCategoryIds = new BookDtoWithoutCategoryIds();
+        bookDtoWithoutCategoryIds.setId(book.getId());
+        bookDtoWithoutCategoryIds.setTitle(book.getTitle());
+        bookDtoWithoutCategoryIds.setAuthor(book.getAuthor());
+        bookDtoWithoutCategoryIds.setIsbn(book.getIsbn());
+        bookDtoWithoutCategoryIds.setPrice(book.getPrice());
+        bookDtoWithoutCategoryIds.setDescription(book.getDescription());
+        bookDtoWithoutCategoryIds.setCoverImage(book.getCoverImage());
+    }
+
+    @Test
+    @DisplayName("Save book - Valid book")
+    public void save_ValidBook_ReturnSavedBookDto() {
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto();
+        createBookRequestDto.setTitle("Test Title");
+        createBookRequestDto.setAuthor("Test Author");
+        createBookRequestDto.setIsbn("978-3-16-148410-0");
+        createBookRequestDto.setPrice(new BigDecimal("19.99"));
+        when(bookMapper.toModel(createBookRequestDto)).thenReturn(book);
```

Comment:
To make test more readable separate with empty lines precondition, action and assertion